src/main/webapp/scripts/tests/comment-post-test.js

```diff
@@ -0,0 +1,77 @@
+import { postVideoComment } from '/scripts/post-comment.js';
+
+const VIDEO_ID = 'sample video id';
+const COMMENT_TEXT = 'sample comment';
+const REQUEST_BODY = {
+    'path': '/youtube/v3/commentThreads?part=snippet',
+    'method': 'POST',
+    'body': {
+        "snippet": {
+            "videoId": VIDEO_ID,
+            "topLevelComment": {
+                "snippet": {
+                    "textOriginal": COMMENT_TEXT,
+                },
+            },
+        },
+    },
+};
+
+QUnit.module('Post a comment', {
+    before: loadClientAsync,
+    beforeEach: () => { sinon.stub(gapi.client, 'request'); },
+    afterEach: () => { gapi.client.request.restore(); },
+});
+
+QUnit.test('Throw an error', function (assert) {
+    var done = assert.async();
+    const errorMsg = 'error message';
+
+    gapi.client.request.withArgs(REQUEST_BODY).returns(new Promise(() => {
+        throw new CustomError(errorMsg);
+    }));
+
+    postVideoComment(VIDEO_ID, COMMENT_TEXT)
+        .catch(error => {
+            assert.equal(error, errorMsg, 'Pass the error message to the callback function');
+            done();
+        });
+});
+
+QUnit.test('Successful post', function (assert) {
+    var done = assert.async();
+
+    gapi.client.request.withArgs(REQUEST_BODY).returns(new Promise((resolve) => {
+        resolve(new CustomResponse());
+    }));
+
+    postVideoComment(VIDEO_ID, COMMENT_TEXT)
+        .then(data => {
+            assert.ok(true, `Don't care about the message`);
```

Comment:
I guess this line is not needed. All that matters is that the the promised was resolved thus just reaches done()

Reply:
This is the error I get if I remove this line: 
`Expected at least one assertion, but none were run - call expect(0) to accept zero assertions.` I will call `expect(0)` instead

Reply:
Done